### PR TITLE
Bugfix/new donation flow issues

### PR DESCRIPTION
**Pull request by nikolay-yankov**
## Motivation and context
Fixes several issues from ticket #1960 related to the new donation flow, discovered during QA testing.

## Screenshots:

Issue #|Before|After
-|---|---
1|image|image
2|image|image
3|Cannot reproduce - message sent to QA for details| N/A
4|image|imageimage
5|image| Yes, it should work exactly the opposite way. Now when the user selects the `X` button or `Cancel` button, the user remains on the same page. If the user selects `Continue` button, then it will be redirected to the Campaign preview page.
10|image|image
11|image|image
12|image|image

### Additional change:
IMO we should not disable the `Amount` field when bank payment option is selected as this creates several side effect. IMO we should either hide it completely when Bank options is chosen or leave it always available. This is because currently, if the user selects an amount and then selects bank payment type, the `Amount` field becomes disabled. This could create some confusion because the user cannot change the amount anymore and they cannot use the calculator for amount + fees at the bottom. Also if validation is triggered and the field, they cannot select an option to continue with the donation.
Here I removed the option for disabling in commit https://github.com/podkrepi-bg/frontend/commit/2ceeef55f2826e3ac9e43e81b25c8d28021b1d9e

### CODE REVIEW

File: src/components/client/donation-flow/DonationFlowForm.tsx [hunk 1]
```diff
@@ -247,10 +247,7 @@ export function DonationFlowForm() {
               </Button>
               <Grid2 container size={{ xs: 12, md: 7 }} direction={'column'}>
                 <StepSplitter content="1" active={Boolean(values.amountChosen)} />
-                <Amount
-                  disabled={values.payment === DonationFormPaymentMethod.BANK}
-                  error={Boolean(errors.finalAmount) && Boolean(submitCount > 0)}
-                />
+                <Amount error={Boolean(errors.finalAmount) && Boolean(submitCount > 0)} />
```
Comment (sashko9807) [inline src/components/client/donation-flow/DonationFlowForm.tsx:250]: Can we please revert this change for now? 
I understand the reasoning for this, but since this Amount Picker has no effect on bank transactions, in my opinion it will cause more confusion to keep it enabled.
I've already contacted the design team to look at this. If there are any changes regarding this step, lets handle them in a separate PR.

**Review by sashko9807 — CHANGES_REQUESTED** [review]

File: src/components/client/donation-flow/DonationFlowForm.tsx (hunk 1 above)
Comment (nikolay-yankov) [reply src/components/client/donation-flow/DonationFlowForm.tsx]: Yes, I just reverted this one. Also reverted the change in `Amount` validation to not take into account the bank type payment with this commit https://github.com/podkrepi-bg/frontend/pull/1962/commits/ef26ae382f30aa2ee2449d33615c90645c090128
Here is a visual what I meant that the user cannot select another amount when Bank is selected and the calculation at the bottom cannot be used if the user wants to see what the fees would be.
cc @swolf86

chrome_WnBgUR0833

Comment (swolf86) [reply src/components/client/donation-flow/DonationFlowForm.tsx]: @nikolay-yankov 
I think the transactions fees should not be displayed it should be ---- until we have the integration with paysera. 

Why the amount cannot be changed, after selecting a bank transfer?
I saw that too

**Review by sashko9807 — APPROVED** [review]
